server/controllers/socketController.js

```diff
@@ -31,15 +31,26 @@ exports.handleJoinChat = (socket) => {
 
 exports.handleMessage = (socket) => {
   socket.on('new_message', (message) => {
-    let chat = message.chat;
-    if (!chat.users) {
-      return console.log('no users');
+    const { chat, sender } = message;
+
+    // Ensure the message and chat have necessary properties
+    if (!chat || !chat.users || !sender) {
+      console.log('Invalid message format');
+      return;
     }
-    chat.users.forEach((user) => {
-      if (user._id === message.sender._id) {
-        return;
+
+    // Use Set to store unique user IDs
+    const recipientUserIds = new Set(chat.users.map((user) => user._id));
+
+    // Remove the sender's ID from the recipients
+    recipientUserIds.delete(sender._id);
+
+    // Broadcast the message to all recipients
+    recipientUserIds.forEach((userId) => {
+      // Check if the user is connected before emitting the message
+      if (connectedUsers.has(userId)) {
```

Comment:
`connectedUsers` is an `Array`; not a `Set`. JavaScript arrays don't have any method named `has()`.

Did you test the code?
I think it would throw an error.

Reply:
Oh, yes, it's an array and I will update this

Reply:
Hmm, so this means you didn't run the project to check if everything was right :(

This is a bad practice. Even if it is a small change, you must ensure things don't break.

Reply:
Setting up certain projects locally can be challenging, and manual changes may seem like a quick solution  that's why I do that. Next time, I will set it up locally even for small changes